bmc.go

```diff
@@ -23,36 +23,55 @@ var (
 	namespace = "bmc"
 )
 
+type Config struct {
```

Comment:
Should this have a more descriptive name? E.g., `V2SessionlessTransportConfig` or `DialConfig`.

Reply:
I agree Config is a little too generic. I would use DialConfig and DialConfigOption. I'm not sure what the conventions are around exporting DialConfig - can we get away with dialConfig?

Reply:
> can we get away with dialConfig?

I believe we can: https://pkg.go.dev/golang.org/x/net/idna#Option